src/js/locks.js

```diff
@@ -52,7 +52,8 @@ define(function (require, exports, module) {
         JS_STYLE: "jsStyle",
         JS_LIBRARIES: "jsLibraries",
         CC_LIBRARIES: "ccLibraries",
-        OS_CLIPBOARD: "osClipboard"
+        OS_CLIPBOARD: "osClipboard",
+        PS_GENERATOR: "psGenerator"
```

Comment:
Assuming your plugin is written to be able to handle multiple concurrent requests, the lock is not necessary. If you're not sure if your plugin is written to handle multiple concurrent requests, we should chat about that offline. The only potential downside to using the lock is, potentially, somewhat worse performance.